### DISCUSSION TITLE
[@types/marked] parameters shouldn't be mandatory

### DISCUSSION THREAD

**Discussion by sernaferna** [DefinitelyTyped/DefinitelyTyped#61664] [Issues with a @types package]
In **@types/marked** some optional parameters are being treated as mandatory.

Including only the relevant bits that I'm aware of:

```typescript
    class Parser {
      parseInline(src: Token[], renderer: Renderer): string; // renderer should be ?
    }

    class Lexer {
      inlineTokens(src: string, tokens: Token[]): Token[]; //tokens should be ?
      inline(src: string, tokens: Token[]): void; // not sure if tokens should be ? or mandatory
    }

```

[a comment by an automated account is not shown]

**Comment by UziTech** [comment]
I believe `Lexer.inline` currently requires a second parameter however it would be better to default the second parameter to `[]` like `inlineTokens`.

If you would like to create a PR in marked for that change and a PR in this repo for the types changes we can get it sorted out.

**Reply by sernaferna** [reply]
Finally created a PR on this repo for the change, after reminding myself how to do it. 🙂
